cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```
@@ -86,6 +84,9 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
     }
 
     ScheduleSpecification spec = appSpec.getSchedules().get(scheduleName);
+    if (!requirementsChecker.checkSatisfied(programId, spec.getSchedule())) {
+      return Futures.<Void>immediateFuture(null);
```

should we throw an exception instead?